File: benches/fixtures/sample.rb
require 'cgi'

def get_env_value
  ENV['APP_SECRET'] || ''
end

def sanitize_html(input)
  CGI.escapeHTML(input)
end

def execute_command(cmd)
  system(cmd)
end

def safe_flow
  val = get_env_value
  clean = sanitize_html(val)
  puts clean
end

def unsafe_flow
  val = get_env_value
  execute_command(val)
end

safe_flow
unsafe_flow

File: benches/fixtures/sample.go
package main

import (
	"fmt"
	"os"
	"os/exec"
	"html"
)

func getEnv() string {
	return os.Getenv("APP_SECRET")
}

func sanitizeHTML(input string) string {
	return html.EscapeString(input)
}

func runCommand(cmd string) {
	exec.Command("sh", "-c", cmd).Run()
}

func safeFlow() {
	val := getEnv()
	clean := sanitizeHTML(val)
	fmt.Println(clean)
}

func unsafeFlow() {
	val := getEnv()
	runCommand(val)
}

func main() {
	safeFlow()
	unsafeFlow()
}
